framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Repositories/BasicRepositoryBase.cs

```diff
@@ -34,6 +34,9 @@ public abstract class BasicRepositoryBase<TEntity> :
 
     public ICancellationTokenProvider CancellationTokenProvider => LazyServiceProvider.LazyGetService<ICancellationTokenProvider>(NullCancellationTokenProvider.Instance);
 
+
+    public bool IsReadOnly { get; set; } = false;
```

Comment:
You've used `ObjectHelper.TrySetProperty` above. So, you can define the setter as `protected` here to prevent changing the value out of the repository class.

```suggestion
    public bool IsReadOnly { get; potected set; }
```

It is already false by default, no need to set to false explicitly.